src/main/java/com/rapid7/client/dcerpc/mslsad/LocalSecurityAuthorityService.java

```diff
@@ -117,10 +119,36 @@ public void closePolicyHandle(final byte[] handle) throws IOException {
         callExpect(request, "LsarClose", SystemErrorCode.ERROR_SUCCESS, SystemErrorCode.STATUS_INVALID_HANDLE);
     }
 
-    public LsarLookupNamesResponse lookupNames(final byte[] policyHandle, final String... names)
+    /**
+     *
+     * @param policyHandle Handle to the policy
+     * @param names Names to lookup SIDs for
+     * @return Returns a list of SIDs
+     * @throws IOException Thrown if exception happens at the RPC layer
+     */
+    public SID[] lookupNames(final byte[] policyHandle, final String... names)
             throws IOException {
         final LsarLookupNamesRequest request = new LsarLookupNamesRequest(parseHandle(policyHandle), names);
-        return callExpectSuccess(request, "LsarLookupNames");
+        LsarLookupNamesResponse response = callExpect(request, "LsarLookupNames",
+                SystemErrorCode.ERROR_SUCCESS,
+                SystemErrorCode.STATUS_SOME_NOT_MAPPED);
+        LSAPRTranslatedSID[] translatedSIDs = response.getLsaprTranslatedSIDs().getLsaprTranslatedSIDArray();
```

Comment:
Can these ever be null (i.e. were they pointers?). If so, then null check and initialize to an empty array (if your service layer doesn't care)

Reply:
Should I catch NPE in my catch block as well for this case?

Reply:
I don't think so. I would just initialize an empty array and carry on. But you only need to do this in the event that what you unmarshalled was a pointer, otherwise it's impossible to be null. Point being, just do a quick check for every object and make the determination on a case-by-case basis.

Reply:
@skhare-r7 Is it a conformant array of reference to SIDs? If so you could use the RPCSIDArray class